sycl/source/detail/kernel_program_cache.hpp

```diff
@@ -82,21 +79,19 @@ class KernelProgramCache {
     return {MKernelsPerProgramCache, MKernelsPerProgramCacheMutex};
   }
 
-  template <class Predicate> void waitUntilBuilt(Predicate Pred) const {
-    std::unique_lock<std::mutex> Lock(MBuildCVMutex);
-
-    MBuildCV.wait(Lock, Pred);
+  template <class Predicate>
+  void waitUntilBuilt(std::condition_variable &CV,
+                      std::unique_lock<std::mutex> &Lock,
+                      Predicate Pred) const {
+    CV.wait(Lock, Pred);
   }
```

Comment:
Also, as far as I see you can leave the contents of this function as it is, i.e. lock the appropriate `mutex` for `BuildResult` and await on its `condition_variable`. I mean, I can't find changes in `program_manager.cpp` which require this kind of change of signature of the function.